Makefile

```diff
@@ -154,27 +157,76 @@ stylus_test_hostio-test_src       = $(call get_stylus_test_rust,hostio-test)
 
 stylus_test_wasms = $(stylus_test_keccak_wasm) $(stylus_test_keccak-100_wasm) $(stylus_test_fallible_wasm) $(stylus_test_storage_wasm) $(stylus_test_multicall_wasm) $(stylus_test_log_wasm) $(stylus_test_create_wasm) $(stylus_test_math_wasm) $(stylus_test_sdk-storage_wasm) $(stylus_test_erc20_wasm) $(stylus_test_read-return-data_wasm) $(stylus_test_evm-data_wasm) $(stylus_test_hostio-test_wasm) $(stylus_test_bfs:.b=.wasm)
 stylus_benchmarks = $(wildcard $(stylus_dir)/*.toml $(stylus_dir)/src/*.rs) $(stylus_test_wasms)
+CBROTLI_WASM_BUILD_ARGS ?=-d
+
+
+ESPRESSO_NETWORK_GO_VER ?= 0.0.36
+ESPRESSO_TAR = espresso-network-go-$(ESPRESSO_NETWORK_GO_VER).tar.gz
+ESPRESSO_URL = https://github.com/EspressoSystems/espresso-network-go/archive/refs/tags/v$(ESPRESSO_NETWORK_GO_VER).tar.gz
+ESPRESSO_DIR = espresso-network-go
 
-espresso_crypto_dir = ./espressocrypto/lib/espresso-crypto-helper
+# Download the tarball
+$(ESPRESSO_TAR):
+	curl -L -o $@ $(ESPRESSO_URL)
+
+# Extract into target directory (strip the top-level folder)
+$(ESPRESSO_DIR): $(ESPRESSO_TAR)
+	@echo "Extracting $(ESPRESSO_TAR) into $(ESPRESSO_DIR)/..."
+	rm -rf $(ESPRESSO_DIR)
+	mkdir -p $(ESPRESSO_DIR)
+	tar -xzf $(ESPRESSO_TAR) --strip-components=1 -C $(ESPRESSO_DIR)
+
+espresso_crypto_dir = $(ESPRESSO_DIR)/verification/rust
 espresso_crypto_files = $(wildcard $(espresso_crypto_dir)/*.toml $(espresso_crypto_dir)/src/*.rs)
 espresso_crypto_lib = $(output_root)/lib/libespresso_crypto_helper
 espresso_crypto_filename = libespresso_crypto_helper.so
-ifeq ($(UNAME_S), Darwin)
+espresso_target_lib = $(ESPRESSO_DIR)/target/lib
+
+
+# Normalize architecture names
+ifeq ($(UNAME_M),arm64)
+    # Apple Silicon reports as arm64, but Rust uses aarch64
+    DETECTED_ARCH := aarch64
+else
+    DETECTED_ARCH := $(UNAME_M)
+endif
+
+# Determine target triple
+ifeq ($(DETECTED_ARCH),aarch64)
+    ifeq ($(UNAME_S),Darwin)
+        TRIPLE := aarch64-apple-darwin
+    else
+        TRIPLE := aarch64-unknown-linux-gnu
+    endif
+else ifeq ($(DETECTED_ARCH),x86_64)
+    ifeq ($(UNAME_S),Darwin)
+        TRIPLE := x86_64-apple-darwin
+    else
+        TRIPLE := x86_64-unknown-linux-gnu
+    endif
+else
+    $(error Architecture $(DETECTED_ARCH) is not supported)
+endif
+
+# Set library extension based on OS
+ifeq ($(UNAME_S),Darwin)
+    LIB_EXT := dylib
 	espresso_crypto_filename = libespresso_crypto_helper.dylib
 else
+    LIB_EXT := so
 	export LD_LIBRARY_PATH := $(shell pwd)/target/lib:$LD_LIBRARY_PATH
 endif
 
-CBROTLI_WASM_BUILD_ARGS ?=-d
 
 # user targets
 .PHONY: build-espresso-crypto-lib
-build-espresso-crypto-lib: $(espresso_crypto_lib)
-
-$(espresso_crypto_lib): $(DEP_PREDICATE) $(espresso_crypto_files)
+build-espresso-crypto-lib: $(ESPRESSO_DIR)
 	mkdir -p `dirname $(espresso_crypto_lib)`
 	cargo build --release --manifest-path $(espresso_crypto_dir)/Cargo.toml
```

Comment:
Since the `espressocryto` has been removed. These lines should fail. Can you check that a bit?

Reply:
espresso_crypto_dir is now pointing to the newly downloaded code from tar file